tests/test_netsync_data_buffer.h

```diff
@@ -321,48 +338,105 @@ TEST_CASE("[Modules][DataBuffer] Vector2") {
 		CHECK_MESSAGE(added_value.x == doctest::Approx(value.x).epsilon(epsilon), "Added Vector2 should have the same x axis");
 		CHECK_MESSAGE(added_value.y == doctest::Approx(value.y).epsilon(epsilon), "Added Vector2 should have the same y axis");
 
+		CHECK(buffer.get_bit_offset() == buffer.get_bit_taken(DataBuffer::DATA_TYPE_VECTOR2, compression_level));
+
 		buffer.begin_read();
 		const Vector2 read_value = buffer.read_vector2(compression_level);
 		CHECK_MESSAGE(read_value.x == doctest::Approx(value.x).epsilon(epsilon), "Read Vector2 should have the same x axis");
 		CHECK_MESSAGE(read_value.y == doctest::Approx(value.y).epsilon(epsilon), "Read Vector2 should have the same y axis");
 	}
 }
 
-TEST_CASE("[Modules][DataBuffer] Vector3") {
+TEST_CASE("[NetSync][DataBuffer] Normalized Vector2") {
+	DataBuffer::CompressionLevel compression_level = DataBuffer::COMPRESSION_LEVEL_3;
+	double epsilon = 0.0;
+
+	SUBCASE("[NetSync][DataBuffer] Compression level 3") {
+		compression_level = DataBuffer::COMPRESSION_LEVEL_3;
+		epsilon = 0.033335;
```

Comment:
should those epsilon values be hard coded like that in the test?
Like I imagine you have those constants somewhere in the NetSync, when we will change those values there the tests will no longer reflect the way how the module is working...

Reply:
These epsilon are calculated according the the compression level used. So it's something you don't want to tweak here

Reply:
I'm more worried about possible discrepancy between test and the actual runtime values

Reply:
Ok, let me reason about this

Reply:
We discussed this privately, but leaving the answer here for others: Since on this test we are comparing the `uncompressed` value with the `compressed` one, we need an epsilon that takes into account the compression precision loss.

We don't need that on the net sync side because we never run such kind of comparison: We always compare uncompressed vars with uncompressed vars or compressed vars with compressed vars.